Pre production

ProdanSergey opened the pull request:
Day 3:

1. Solve API fetch on server;
2. Solve MongoDB connection;
3. Fetch data from front-end to DB;
4. Working on front-end validation;

ProdanSergey committed:
day2

Registration form added. Working on react-router

Issue 1: Can't solve problem with api

ProdanSergey committed:
Day 3. Registration form serialized

ProdanSergey committed:
Day 3. Validation

ProdanSergey committed:
Day3. Touched fields

ProdanSergey committed:
Day 3. Final

sturchaninov commented on client/src/components/Navigation.js:
```diff
@@ -1,26 +1,22 @@
                             <ul className="navbar-nav mr-auto">
                                 <li className="nav-item">
-                                    <a className="nav-link"><Link to='/registration'>Registration</Link></a>
+                        <Link to='/registration' className="nav-link">Registration</Link>
```
misaligned

sturchaninov commented on client/src/components/Registration.js:
```diff
@@ -1,32 +1,132 @@
+    const target = event.target;
+    const value = target.value;
+    const name = target.name;
+    regex.test(value) ? 
```
maybe would be better to replace with this.setState({inputValid: {...this.state.inputValid, [name]: regex.test(value)}}) ?

ProdanSergey committed:
Day 3. Redux

- create Redux store
- create rootReducer
- add Actions
- add User Add reducer
- fixed text inputs validation

store new user data and ifRegSuccess flag on ReduxStore
